cli/collection_create.go

```diff
@@ -17,25 +17,34 @@ import (
 	"github.com/spf13/cobra"
 
 	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/datastore"
 	"github.com/sourcenetwork/defradb/internal/db"
+	"github.com/sourcenetwork/defradb/internal/encryption"
 )
 
 func MakeCollectionCreateCommand() *cobra.Command {
 	var file string
 	var shouldEncrypt bool
+	var encryptedFields []string
 	var cmd = &cobra.Command{
-		Use:   "create [-i --identity] [-e --encrypt] <document>",
+		Use:   "create [-i --identity] [-e --encrypt] [--encrypt-fields] <document>",
 		Short: "Create a new document.",
 		Long: `Create a new document.
 		
 Options:
-    -i, --identity 
-        Marks the document as private and set the identity as the owner. The access to the document
+	-i, --identity 
+		Marks the document as private and set the identity as the owner. The access to the document
 		and permissions are controlled by ACP (Access Control Policy).
 
 	-e, --encrypt
 		Encrypt flag specified if the document needs to be encrypted. If set, DefraDB will generate a
 		symmetric key for encryption using AES-GCM.
+	
+	--encrypt-fields
+		Comma-separated list of fields to encrypt. If set, DefraDB will encrypt only the specified fields
+		and for every field in the list it will generate a symmetric key for encryption using AES-GCM.
+		If combined with '--encrypt' flag, all the fields in the document not listed in '--encrypt-fields' 
+		will be encrypted with the same key.
```

Comment:
question: Just need some clarification, when you say "will be encrypted with the same key" you mean that all the `not listed` fields will share a key that will be different from the per field key generated for the "listed fields specified by --encrypt-fields"?

Reply:
yes